subsys/net/ip/dhcpv4.c

```diff
@@ -1236,6 +1276,30 @@ static void dhcpv4_start_internal(struct net_if *iface, bool first_start)
 	k_mutex_unlock(&lock);
 }
 
+int net_dhcpv4_add_option_callback(struct net_dhcpv4_option_callback *cb)
+{
+	__ASSERT(cb, "No callback!");
+	__ASSERT(cb->handler, "No callback handler!");
+
+	sys_slist_prepend(&option_callbacks, &cb->node);
+	return 0;
+}
+
+int net_dhcpv4_remove_option_callback(struct net_dhcpv4_option_callback *cb)
+{
+	__ASSERT(cb, "No callback!");
+	__ASSERT(cb->handler, "No callback handler!");
+
+	if (!sys_slist_is_empty(&option_callbacks)) {
```

Comment:
`sys_slist_is_empty()`  seems redundant here, you could just call `sys_slist_find_and_remove()` directly - if the list is empty, it'll fail anyway.

Reply:
As you noticed above, I started with the gpio_callback as reference. That's where this originates from. I'll simplify it.